babel/src/test/java/org/apache/calcite/test/BabelParserTest.java

```diff
@@ -431,6 +431,60 @@ private void checkParseInfixCast(String sqlType) {
     sql(sql).ok(expected);
   }
 
+  @Test public void testDateTimePrimaryColumn() {
+    final String sql = "select time_column";
+    final String expected = "SELECT `TIME_COLUMN`";
+    sql(sql).ok(expected);
+  }
+
+  @Test public void testDateTimePrimaryLiteral() {
+    final String sql = "select timestamp '2020-05-30 13:20:00'";
+    final String expected = "SELECT TIMESTAMP '2020-05-30 13:20:00'";
+    sql(sql).ok(expected);
+  }
+
+  @Test public void testDateTimePrimaryBuiltInFunction() {
+    final String sql = "select current_timestamp";
+    final String expected = "SELECT CURRENT_TIMESTAMP";
+    sql(sql).ok(expected);
+  }
+
+  @Test public void testDateTimePrimaryLiteralAtLocal() {
+    final String sql = "select timestamp '2020-05-30 13:20:00' at local";
+    final String expected = "SELECT TIMESTAMP '2020-05-30 13:20:00' AT LOCAL";
+    sql(sql).ok(expected);
+  }
+
+  @Test public void testDateTimePrimaryColumnAtLocal() {
+    final String sql = "select time_column at local";
+    final String expected = "SELECT `TIME_COLUMN` AT LOCAL";
+    sql(sql).ok(expected);
+  }
+
+  @Test public void testDateTimePrimaryBuiltInFunctionAtLocal() {
+    final String sql = "select current_timestamp at local";
+    final String expected = "SELECT CURRENT_TIMESTAMP AT LOCAL";
+    sql(sql).ok(expected);
+  }
+
+  @Test public void testDateTimePrimaryLiteralAtTimeZone() {
+    final String sql = "select timestamp '2020-05-30 13:20:00' at time zone \"GMT+10\"";
+    final String expected = "SELECT TIMESTAMP '2020-05-30 13:20:00' AT TIME ZONE `GMT+10`";
+    sql(sql).ok(expected);
+  }
+
+  @Test public void testDateTimePrimaryColumnAtTimeZone() {
+    final String sql = "select time_column at time zone \"Europe Moscow\"";
+    final String expected = "SELECT `TIME_COLUMN` AT TIME ZONE `Europe Moscow`";
+    sql(sql).ok(expected);
+  }
+
+  @Test public void testDateTimePrimaryBuiltInFunctionAtTimeZone() {
+    final String sql = "select current_date at time zone \"GMT-5\"";
+    final String expected = "SELECT CURRENT_DATE AT TIME ZONE `GMT-5`";
+    sql(sql).ok(expected);
+  }
+
```

Comment:
I noticed in the design docs that it says this syntax can also appear elsewhere besides a "select" query. It might be worthwhile to add a test where this syntax is in a different type of query to show it works outside of a select statement

Reply:
Excellent suggestion! Will add more test case to show that this would work in a where clause.

Reply:
Changes made in https://github.com/googleinterns/calcite/pull/51/commits/f542d9676b99e435baa4fdf824909c6638afc482